allennlp/interpret/attack/input_reduction.py

```diff
@@ -0,0 +1,123 @@
+from typing import Dict, List, Set
+import numpy as np
+import torch
+from allennlp.interpret.attack import Attacker
+from allennlp.common.util import JsonDict, sanitize 
+from allennlp.data import Instance
+from allennlp.predictors.predictor import Predictor
+from collections import defaultdict
+
+@Attacker.register('input-reduction')
+class InputReduction(Attacker):
+    """
+    Runs the input reduction method from https://arxiv.org/abs/1804.07781.
```

Comment:
When you have links to arxiv, it'd be nice to also give at least a title, maybe also abbreviated author list, so I don't have to click the click to know what paper you're referring to.